### PR TITLE
Added missing WebhookClient

### PR DESCRIPTION
**Pull request by PurgePJ**

### CODE REVIEW

File: libs/client/API.lua [hunk 1]
```diff
@@ -657,37 +657,37 @@ function API:getWebhookWithToken(webhook_id, webhook_token) -- not exposed, need
 	return self:request("GET", endpoint)
 end
 
-function API:modifyWebhook(webhook_id, payload) -- Webhook:_modify
+function API:modifyWebhook(webhook_id, payload) -- WebhookClient:modify
```
Comment (SinisterRectus) [inline libs/client/API.lua]: Please undo this change.

File: libs/client/API.lua [hunk 2]
```diff
@@ -657,37 +657,37 @@ function API:getWebhookWithToken(webhook_id, webhook_token) -- not exposed, need
...
 end
 
-function API:modifyWebhookWithToken(webhook_id, webhook_token, payload) -- not exposed, needs webhook client
+function API:modifyWebhookWithToken(webhook_id, webhook_token, payload) -- WebhookClient:modifyWebhookWithToken
 	local endpoint = f(endpoints.WEBHOOK_TOKEN, webhook_id, webhook_token)
 	return self:request("PATCH", endpoint, payload)
 end
 
-function API:deleteWebhook(webhook_id) -- Webhook:delete
+function API:deleteWebhook(webhook_id) -- WebhookClient:delete
```
Comment (SinisterRectus) [inline libs/client/API.lua]: Please undo this change.

File: libs/client/WebhookClient.lua [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+function get.name(self)
+	return self._options.name
+end
+
+function get.token(self)
+	return self._options.token
+end
+
+-- Setters and Modifiers
+function WebhookClient:modify(tbl)
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: This should be a private method `_modify`

File: libs/client/WebhookClient.lua [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+	local data, err = self._api:modifyWebhookWithToken(self._options.id, self._options.token, tbl)
+	if data then 
+		return true
+	else
+		return false, err
+	end
+end
+
+function WebhookClient:setAvatar(avatar)
+	avatar = Resolver.base64(avatar) or resolveImage(avatar) or null
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: Why is `resolveImage` used here? `avatar` should be base64 or an image path.

File: libs/client/WebhookClient.lua [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+function get.token(self)
+	return self._options.token
+end
+
+-- Setters and Modifiers
+function WebhookClient:modify(tbl) 
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
+
+	for k, v in pairs(tbl) do
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: It's not necessary to check for channel_id, just ignore it like I do with all of the other PATCHes.

File: libs/client/WebhookClient.lua [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+	return self._options.name
+end
+
+function get.token(self)
+	return self._options.token
+end
+
+-- Setters and Modifiers
+function WebhookClient:modify(tbl) 
+	assert(type(tbl) == "table", "Invalid modify parameter. Table expected")
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: This check is not necessary.

File: libs/client/WebhookClient.lua [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+
+	local data, err = self:modify({avatar = avatar})
+	if data then
+		return true
+	else
+		return nil, err
+	end
+end
+
+function WebhookClient:setChannelId(channelId)
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: I thought you can't change the channel?

File: libs/client/WebhookClient.lua [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+	}, webhookData.file, wait)
+
+	if data then
+		return true
+	else
+		return nil, err
+	end
+end
+
+function WebhookClient:delete()
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: Not necessary to have two methods. Just use `delete`. Kind of a bad name, though. It looks like you're deleting the Client.

File: libs/client/WebhookClient.lua [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+	return self:deleteWithToken()
+end
+
+function WebhookClient:deleteWithToken()
```
Comment (SinisterRectus) [inline libs/client/WebhookClient.lua]: See above.

File: libs/client/WebhookClient.lua (hunk 8 above)
Comment (PurgePJ) [reply libs/client/WebhookClient.lua]: If you were going to delete the client I would use ``destroy``. Delete is used to delete the webhook from discord.

Comment (SinisterRectus) [reply libs/client/WebhookClient.lua]: Maybe `deleteWebhook` to indicate that you are actually deleting the webhook and not the client. I do not like "destroy".

Comment (SinisterRectus) [reply libs/client/WebhookClient.lua]: But then `setName` and `setAvatar` become confusing because you're setting the properties of the webhook, not the client. Hmmm.

Comment (PurgePJ) [reply libs/client/WebhookClient.lua]: In the end we need to have in mind that we are managing a webhook using an imitation of a Client. I will change ``delete`` to ``deleteWebhook``

**Comment by SinisterRectus** [comment]
I will address this after voice is mostly completed.

**Comment by trumpCoder** [comment]
I guess this should be added... Other libraries have this.

**Comment by trumpCoder** [comment]
Please add this in. Today
